### PR TITLE
Fix the broken install of the azure-cli and invalid availabilitySet parameters

### PR DESCRIPTION
**Pull request by gavindidrichsen**

### CODE REVIEW

**Review by russellseymour — APPROVED** [review]
This looks good 👍 

I would suggest making the `availbilitySet` values configureable through the parameters but they have a default value of 2 set. This can be another tweak later on.

File: azuredeploy.json [hunk 1]
```diff
@@ -225,8 +225,8 @@
...
...'] = ''batch''\n','\n','bookshelf[''storage_type''] = :sql\n ','\n','rabbitmq[''enable''] = false\n','rabbitmq[''management_enabled''] = false\n','rabbitmq[''queue_length_monitor_enabled''] = false\n','\n','opscode_expander[''enable''] = false\n','\n','dark_launch[''actions''] = false\n','\n','opscode_erchef[''nginx_bookshelf_caching''] = :on\n','opscode_erchef[''s3_url_expiry_window_size''] = ''50%''\n','opscode_erchef[''s3_url_expiry_window_size''] = ''100%''\n','license[''nodes''] = 999999\n','oc_chef_authz[''http_init_count''] = 100\n','oc_chef_authz[''http_max_count''] = 100\n','oc_chef_authz[''http_queue_max''] = 200\n','oc_bifrost[''db_pool_size''] = 20\n','oc_bifrost[''db_pool_queue_max''] = 40\n','oc_bifrost[''db_pooler_timeout''] = 2000\n','opscode_erchef[''depsolver_worker_count''] = 4\n','opscode_erchef[''depsolver_timeout''] = 20000\n','opscode_erchef[''db_pool_size''] = 20\n','opscode_erchef[''db_pool_queue_max''] = 40\n','opscode_erchef[''db_pooler_timeout''] = 2000\n','opscode_erchef[''authz_pooler_timeout''] = 2000\n','EOF\n','sed -i ''0,/######/s//''$dbPassword''/'' /etc/opscode/chef-server.rb\n','sed -i ''0,/####/s//''$fqdn''/'' /etc/opscode/chef-server.rb\n')]",
```
Comment (skpaterson) [inline azuredeploy.json:228]: Might it be worth writing a script to populate this JSON based on a placeholder value?  For example:
```
"chef_frontend_install_script_base": "[concat(FRONTEND_SCRIPT)]"
"chef_automate_install_script_base": "[concat(AUTOMATE_SCRIPT)]"
```
That way the scripts themselves could live separately in more easily managed files.

**Review by skpaterson — APPROVED** [review]
Agree with @russellseymour's comments.  Looks good!

File: azuredeploy.json (hunk 1 above)
Comment (russellseymour) [reply azuredeploy.json:228]: @skpaterson, @gavindidrichsen is going to break the scripts out of being inline into their own files
The template will then pull them in as it does for the other nested templates

Comment (gavindidrichsen) [reply azuredeploy.json:228]: @skpaterson, @russellseymour:  Added both your suggestions to the issues list so they can get picked up later.

Comment (skpaterson) [reply azuredeploy.json:228]: 👍

**Pull request merged**